src/common/cockpitwebserver.c

```diff
@@ -883,7 +883,7 @@ parse_and_process_request (CockpitRequest *request)
   guint64 length;
 
   /* The hard input limit, we just terminate the connection */
-  if (request->buffer->len > cockpit_webserver_request_maximum * 2)
+  if (request->buffer->len > cockpit_webserver_request_maximum)
```

Comment:
Probably it makes sense now to move this check after the read.

Reply:
Fully agreed, I'll clean this up in a follow-up.